Bottom: 250d790e7a782b4f26bbd2935aef417ad843968f
Top:    c9a319d0561da4be5a6f636310eed1e9d4446b7d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-24 02:40:31 +0100

Refresh of draw-c-change-drawsilk-to-use-

---

diff --git a/src/draw.c b/src/draw.c
index 1cbc98b..381ef1d 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -309,9 +309,9 @@ rat_callback (const BoxType * b, void *cl)
  */
 
 static void
-PrintAssembly (const BoxType * drawn_area, int side_group, int swap_ident)
+PrintAssembly (int side, const BoxType * drawn_area)
 {
-  int save_swap = SWAP_IDENT;
+  int side_group = GetLayerGroupNumberByNumber (max_copper_layer + side);
 
   gui->set_draw_faded (Output.fgGC, 1);
   DrawLayerGroup (side_group, drawn_area);
@@ -493,11 +493,11 @@ DrawEverything (BoxTypePtr drawn_area)
     }
 
   doing_assy = true;
-  if (gui->set_layer ("topassembly", SL (ASSY, TOP), 0))
-    PrintAssembly (drawn_area, component, 0);
+  if (gui->set_layer ("topassembly", SL (ASSY, TOP)))
+    PrintAssembly (COMPONENT_LAYER, drawn_area, component, 0);
 
   if (gui->set_layer ("bottomassembly", SL (ASSY, BOTTOM), 0))
-    PrintAssembly (drawn_area, solder, 1);
+    PrintAssembly (SOLDER_LAYER, drawn_area, solder);
   doing_assy = false;
 
   if (gui->set_layer ("fab", SL (FAB, 0), 0))
